### DISCUSSION TITLE
RX-A830 works!

### DISCUSSION THREAD

**Discussion by phazuzu** [mvdwetering/yamaha_ynca#240] [General]
Hi guys,

just want to let you know that a RX-A830 just works beautifully ;)

Thanks a lot for your effort,

phazuzu

**Comment by mvdwetering** [comment]
Cool, thanks for the update 👍 
Added it to the list in the readme

**Comment by matthiasbeyer** [comment]
Hey @phazuzu - I have a A830 as well... mine sometimes loses connection to home assistant (or home assistant loses connection to the receiver).
Apparently (I did a quick `telnet` test) the port gets closed at some point...

Although I can ping its IP, the ynca integration in home-assistant does not find it and it is "unavailable" in HA. Do you have the same problem and - hopefully - a solution?

**Reply by mvdwetering** [reply]
There is a setting on the receiver called "Network standby" which turns off the network when the receiver is in standby. Maybe that settings is Off and it therefore kills the connection? If so try turning it On.

If that does not help please collect debug logging and create a new issue so we can have a better look at it.

Debug logging can be enabled in the integration settings. Bottom option in the screenshot. Just enable it and let it run until the issue happens. After that click the option again (it is now called "Disable debug logging") and the debug logs will be downloaded. Attach that file to the issue. Hopefully it gives some more insights on what is happening.
![image](https://github.com/user-attachments/assets/2af77dc9-eca3-48fc-9759-e3e07315f5c2)

**Reply by phazuzu** [reply]
Hi Matthias,

sorry for my late respone, i was a few days afk...

as in my initial post, the integration works beautifully - Thanks again ;)
but I remember there was something at the beginning, but not exactly what it was... Since then, everything is pretty fine!

I would strongly head for Michels proposal, as the Network Standby Option was switched on on my side!
Right now, I swiched it off - maybe I can reproduce the error.

Did you know that the RX-830 already have a built-in webserver, it may makes some investigations a bit easier ;)

**Reply by matthiasbeyer** [reply]
Thank you both for your replies. The "Network Standby" option is turned on in my receiver. I did not yet come around to debug this further (not enough time). I'll reply here once I got the chance to!

**Reply by matthiasbeyer** [reply]
Okay, so here is what I see:

The receiver is reachable via 192.168.178.37, I recorded this screenshot of the webinterface of the receiver:

![image](https://github.com/user-attachments/assets/a8a67ea9-4a55-4083-82d3-03d39a4d10e9)

But home assistant does not find it via this integration:

![image](https://github.com/user-attachments/assets/b6636a91-1e3c-4653-b7bb-6124783b5c10)

There's no other software that could potentially access that device.

**Reply by mvdwetering** [reply]
Strange, I think that message is only to show for connectivity issues. Maybe your Home Assistant instance can not get to that IP? (I assume the screenshot was taken on a different machine)

Unfortunately I don't really have more detailed logging right now.
I am working on some cleanup and minor fixes and will include additional logging for connectivity issues during setup.
However I think it will take at least until this weekend until I get that done, maybe a week longer.

**Reply by matthiasbeyer** [reply]
> Maybe your Home Assistant instance can not get to that IP? (I assume the screenshot was taken on a different machine)

Yes, that's on a different machine. When I log into my home server, `sudo -u hass bash` to the user that HA runs as, and then `ping 192.168.178.37`, that works perfectly fine. So I assume that connectivity is not really the issue here.

> I am working on some cleanup and minor fixes and will include additional logging for connectivity issues during setup.
> However I think it will take at least until this weekend until I get that done, maybe a week longer.

That sounds great! Feel free to ping me as soon as there's a release, I will then package the update for my distro asap. No worries about the timeline though, my home setup is a process, nothing that I expect to be ready at some point :laughing:

**Reply by mvdwetering** [reply]
Sorry for the delayed reply it took longer than intended to wrap it all up 😞

If you are still in need, now (from v8.0.0) the integration logs the complete exception in the Home Assistant logs on the initial setup. However during my testing with forced connection issues the info that could be obtained from those exceptions was not really helping. So not sure if it will help in your case.

**Comment by UKenGB** [comment]
I am having the same problem with an RX-A810 as Michel (@mvdwetering) knows.

Network Standby is ON and works as it should - for a while. Then although the ethernet interface remains on and responds to ping, the  Yamaha won't respond and so the connection is dropped.

Yamaha's own control app and the AVR's web interface similarly stop functioning when this occurs.

I use Home Assistant/YNCA to monitor when it drops and turn the AVR off then on again (so the network interface is working again) using a smart plug so most of the time it is controllable over the network. I also monitor when this happens and it is several times a day. Turning the AVR on from Standby also wakes up its ethernet interface, but power off and on again does the same thing and is simpler.

I do not believe it is in any way caused by this Integration and I have 2 other Yamaha amps working with YNCA and they NEVER stop responding over the LAN.

It looks to me like Yamaha have a problem with the firmware for some of their AVRs, maybe all the RX-A8xx range have this bug. It would sure be annoying if I bought a different replacement and that had the same problem.
